File: fixtures/consumers/logistics-service/src/shipments_client.go
package logistics

const ShippingBase = "https://api.acme-shipping.com/v1"

func SubmitShipment(request map[string]interface{}) map[string]interface{} {
	// request is a JSON-shaped map pinned to Shipping API v1
	return map[string]interface{}{
		"id":       "shp_1",
		"status":   "queued",
		"carrier":  request["carrier"],
		"weightKg": request["weightKg"],
	}
}

func IsShipmentQueued(record map[string]interface{}) bool {
	return record["status"] == "queued"
}

func IsShipmentDelivered(record map[string]interface{}) bool {
	return record["status"] == "delivered"
}

File: fixtures/consumers/logistics-service/src/shipments_client.py
SHIPPING_BASE = "https://api.acme-shipping.com/v1"


def submit_shipment(request):
    # request is a JSON dict pinned to Shipping API v1
    return {"id": "shp_1", "status": "queued", "carrier": request["carrier"], "weightKg": request["weightKg"]}


def is_shipment_queued(record):
    return record["status"] == "queued"


def is_shipment_delivered(record):
    return record["status"] == "delivered"
